src/quail/ADCTask.cpp

```diff
@@ -1,116 +1,141 @@
+    sys.adc.setMode(Ad7124::ContinuousMode);
+
+    // Set up configurations
+    // Initialize pressure configuration
+    sys.adc.setConfig(PRESSURE_CFG, Ad7124::RefIn2, Ad7124::Pga1, false);
+    sys.adc.setConfigFilter(PRESSURE_CFG, Ad7124::Sinc3Filter, 10);
+
+    // Initialize load configuration
+    sys.adc.setConfig(LOAD_CFG, Ad7124::RefIn1, Ad7124::Pga16, true);
+    sys.adc.setConfigFilter(LOAD_CFG, Ad7124::Sinc3Filter, 10);
+
+    // Initialize thermal configuration
+    sys.adc.setConfig(THERMAL_CFG, Ad7124::RefAVdd, Ad7124::Pga1, true);
+    sys.adc.setConfigFilter(THERMAL_CFG, Ad7124::Sinc3Filter, 10);
 
-    bool led = HIGH;
-    uint32_t time = xTaskGetTickCount();
-    while (true)
+    // Set up sensors
+    SensorTask::initialize(); // static function initializes ADC channels for Sensor list in SensorTask.cpp
+
+    // Set adc timeout
+    sys.adc.setTimeout(1);
+
+    while(true)
     {
-        led = !led;
-        digitalWrite(1, led);
-
-        uint32_t SBraw[6];
-
-        for (int i = 1; i <= 6; i++)
-        {
-            adc.waitThenReadData();
-            uint32_t dataWord;
-            uint8_t channel;
-            adc.getData(dataWord,channel);
-            SBraw[channel] = dataWord;
-            // Serial.print(channel);
-            // Serial.print(" : ");
-            // Serial.println(dataWord);
-        }
-
-        // double voltage = Ad7124Chip::toVoltage(SBraw[0], 16, 2.5, true);
-        // // Print result
-        // voltage = voltage / 5.0;
-        // double load = voltage / 0.030 * 1000.0;
-        // // Serial.print("LC1: ");
-        // // Serial.print(SBraw[0]);
-        // // Serial.print("    ");
-        // // Serial.print(voltage, 5);
-        // // Serial.print("    ");
-        // Serial.println(load, 3);
-
-        // double voltage = Ad7124Chip::toVoltage(SBraw[1], 1, 2.5, false);
-        // voltage = voltage * 1.5;
-        // double pressure = (voltage - 0.5) / 4.0 * 1000;
-        // Serial.print("PT1: ");
-        // Serial.print(SBraw[1]);
-        // Serial.print("    ");
-        // Serial.print(voltage, 3);
-        // Serial.print("    ");
-        // Serial.println(pressure, 3);
-
-        double voltage = Ad7124Chip::toVoltage(SBraw[2], 1, 3.6, true);
-        double temp = voltage / 0.005; // 5mV per degree C
-        // Serial.print("TC1: ");
-        // Serial.print(SBraw[2]);
-        // Serial.print("    ");
-        // Serial.print(voltage, 3);
-        // Serial.print("    ");
-        Serial.println(temp, 3);
-
-        // uint32_t newtime = xTaskGetTickCount();
-        // Serial.println(newtime - time);
-        // time = newtime;
+        // TODO: Make this non-blocking.
+        sys.adc.waitThenReadData();
+        adcdata_t adc_data;
+        sys.adc.getData(adc_data.dataword, adc_data.channel);
+        sys.tasks.sensortask.addADCData(adc_data);
     }
+
+    // //Channel 0: Load Cell 1
```

Comment:
feel free to delete old stuff, we can always go back in git, and it isn't worth keeping

Reply:
oops meant to clear that, I kept it around to quickly reference what you had been doing before without branch-switching